google/cloud/storage/internal/object_write_streambuf.cc

```diff
@@ -75,6 +75,7 @@ bool ObjectWriteStreambuf::ValidateHash(ObjectMetadata const& meta) {
     auto function = std::move(hash_function_);
     hash_values_ = std::move(*function).Finish();
   }
+  if (!hash_validator_) return !hash_validator_result_.is_mismatch;
```

Comment:
not necessary, but should we log some sort of warning here?

Reply:
Thanks.  I am not sure how the warning would help.  I mean, yes, maybe they would see it and stop calling `.Close()` twice. But would it matter now that `.Close()` is idempotent?

Reply:
Ah, point taken.